### PR TITLE
Proximal bugfixes

### PR DESCRIPTION
**Pull request by roussel-ryan**
## Motivation

This PR adds support for ModelListGP models in the Proximal biasing acquisition function. Previously, model indexing errors prevented proper selection of the most recently observed point. ModelListGP models are required for flexible optimization of multi-objective and constrained problems.

Note that we check to make sure that the training inputs for all models are identical

### Have you read the [Contributing Guidelines on pull requests](https://github.com/pytorch/botorch/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan
Unit tests added

## Related PRs
# 919 https://github.com/pytorch/botorch/pull/919#issue-977505704

### CODE REVIEW

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:108]: This could potentially also be some other model, not necessarily `SingleTaskGP`
```suggestion
            # non-ModelListGP model
```

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:84]:
```suggestion
                    model.batch_shape != torch.Size([])
```

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:144]:
```suggestion
        train_inputs = self.acq_func.model.train_inputs[0]
        if isinstance(model, ModelListGP):
            train_inputs = train_inputs[0]
        last_X = train_inputs[-1].reshape(1, 1, -1)
```

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:76]: Would you mind putting this into a helper function (e.g. `_validate_model`) so that the constructor doesn't get excessively long and becomes more readable? 

I also seems that by factoring out some of the checks into another function some of the code duplication could be avoided.

**Review by Balandat — COMMENTED** [review]
Thanks for fixing this! Overall this lgtm, just some comments inline.

**Comment by roussel-ryan** [comment]
Need to make another quick fix, don't merge it just yet please

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:90]:
```suggestion
        if isinstance(model, ModelListGP):
            train_inputs = train_inputs[0]
            model = model.models[0]
        input_transform = _get_input_transform(model)
```

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:105]:
```suggestion
def _validate_model(model: Model, proximal_weights: Tensor) -> None:
```

File: botorch/acquisition/proximal.py
Comment (Balandat) [inline botorch/acquisition/proximal.py:166]:
```suggestion
def _get_input_transform(model: Model) -> Optional[InputTransform]:
```

**Review by Balandat — COMMENTED** [review]
Great, thanks for the edits. A couple more nits and this can go in!

**Comment by roussel-ryan** [comment]
Should be all set

**Review by Balandat — APPROVED** [review]

**Comment by roussel-ryan** [comment]
I'm not sure whats going on with this internal test but it seems to have gone all weekend, just something to note if there is an issue

**Comment by roussel-ryan** [comment]
@Balandat there seems to have been an issue with this after you approved the changes. I'm not sure if it was merged or not

**Comment by Balandat** [comment]
@roussel-ryan no worries these internal test suites are massive and it's quite common that something is flake and show up red on github. I made sure there wasn't anything related to your changes, fixed up a couple of things and landed this. Thanks for your contribution!

**Pull request closed without merging**
